erpnext/regional/italy/utils.py

```diff
@@ -0,0 +1,279 @@
+	customer_address = frappe.get_doc("Address", invoice.customer_address)
+	invoice["customer_address_data"] = customer_address
+
+	if invoice.shipping_address_name:
+		invoice["shipping_address_data"] = frappe.get_doc("Address", invoice.shipping_address_name)
+
+	if invoice["customer_data"].is_public_administration:
+		invoice["transmission_format_code"] = "FPA12"
+	else:
+		invoice["transmission_format_code"] = "FPR12"
+
+
+	items = frappe.get_all("Sales Invoice Item", filters={"parent":invoice.name}, fields=["*"], order_by="idx")
+	taxes = frappe.get_all("Sales Taxes and Charges", filters={"parent":invoice.name}, fields=["*"], order_by="idx")
+	tax_data = get_invoice_summary(items, taxes)
+
+	invoice["tax_data"] = tax_data
+
+	#Check if stamp duty (Bollo) of 2 EUR exists.
+	stamp_duty_charge_row = next((tax for tax in taxes if tax.charge_type == _("Actual") and tax.tax_amount == 2.0 ), None)
+	if stamp_duty_charge_row:
+		invoice["stamp_duty"] = stamp_duty_charge_row.tax_amount
+
+	#append items, and tax exemption reason.
+	for item in items:
+		if item.tax_rate == 0.0:
+			item["tax_exemption_reason"] = tax_data["0.0"]["tax_exemption_reason"]
+
+	invoice["invoice_items"] = items
+
+	invoice["payment_terms"] = frappe.get_all("Payment Schedule", filters={"parent": invoice.name}, fields=["*"])
+
+	return invoice
+
+def get_conditions(filters):
+	filters = json.loads(filters)
+
+	conditions = {"docstatus": 1}
+
+	if filters.get("company"): conditions["company"] = filters["company"]
+	if filters.get("customer"): conditions["customer"] =  filters["customer"]
+
+	if filters.get("from_date"): conditions["posting_date"] = (">=", filters["from_date"])
+	if filters.get("to_date"): conditions["posting_date"] = ("<=", filters["to_date"])
+
+	if filters.get("from_date") and filters.get("to_date"):
+		conditions["posting_date"] = ("between", [filters.get("from_date"), filters.get("to_date")])
+
+	return conditions
+
+#TODO: Use function from frappe once PR #6853 is merged.
+def download_zip(files, output_filename):
+	from zipfile import ZipFile
+
+	input_files = [frappe.get_site_path('private', 'files', filename) for filename in files]
+	output_path = frappe.get_site_path('private', 'files', output_filename)
+
+	with ZipFile(output_path, 'w') as output_zip:
+		for input_file in input_files:
+			output_zip.write(input_file, arcname=os.path.basename(input_file))
+
+	with open(output_path, 'rb') as fileobj:
+		filedata = fileobj.read()
+
+	frappe.local.response.filename = output_filename
+	frappe.local.response.filecontent = filedata
+	frappe.local.response.type = "download"
+
+def cleanup_files(files):
+	#TODO: Clean up XML files after ZIP gets downloaded
+	pass
+
+def extract_doc_number(doc):
+	if not hasattr(doc, "naming_series"):
+		return doc.name
+
+	name_parts = doc.name.split("-")
+
+	if hasattr(doc, "amended_from"):
+		if doc.amended_from:
+			return name_parts[-2:-1][0]
+		else:
+			return name_parts[-1:][0]
+	else:
+		return doc.name
```

Comment:
Also extract_doc_number method has not called from anywhere, is it required?

Reply:
Will drop.